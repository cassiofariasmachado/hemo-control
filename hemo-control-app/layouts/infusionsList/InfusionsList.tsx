import React, { FC, useEffect } from 'react';
import styled from 'styled-components/native';
import { InfusionResponse } from '../../models';
import { FlatList, View, Text } from 'react-native';

type InfusionsProps = {
    infusions: InfusionResponse[]
}

const Container = styled.View`
    flex: 1;
    marginTop: 50px;
    alignItems: center;
    flexDirection: column;
    alignItems: stretch;
`

const ListItem = styled.View`
    padding: 20px;
    margin: 10px;
    borderRadius: 5px;
    backgroundColor: #fff;
    alignItems: stretch;
    alignSelf: stretch;
    justifyContent: center;
    marginHorizontal: 30px;
    height: 80px;
    shadowOpacity: 0.20;
    shadowRadius: 2.20px;
    elevation: 3;
`

const ListItemTitle = styled.Text`
    fontWeight: bold;
`;

const ListItemText = styled.Text`
    color: black;
    fontSize: 16px;
`

export const InfusionsList: FC<InfusionsProps> = ({ infusions }) => {
    const renderItem = (item: InfusionResponse) => (
        <ListItem>
            <ListItemText><ListItemTitle>{item.bleedingLocal}</ListItemTitle></ListItemText>
            <ListItemText>
                <ListItemTitle>Data e local: </ListItemTitle>
                {`${item.treatmentLocal}, ${new Date(item.date).toLocaleDateString()} ${new Date(item.date).toLocaleTimeString()}`}
            </ListItemText>
            <ListItemText>
                <ListItemTitle>Fator: </ListItemTitle>
                {item.factor ? ` ${item.factor.brand}, ${item.factor.unity} UI, Lote ${item.factor.lot}` : 'NÃ£o informado'}
            </ListItemText>
        </ListItem>
    );

    return <Container>
        <FlatList data={infusions} renderItem={({ item }) => renderItem(item)} keyExtractor={item => item.id.toString()} />
    </Container>;
};
