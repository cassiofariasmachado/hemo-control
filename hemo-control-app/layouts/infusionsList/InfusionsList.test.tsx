import React from 'react';
import { create } from 'react-test-renderer';
import { InfusionsList } from './InfusionsList';

describe(InfusionsList.name, () => {
    test('renders correctly', () => {
        const infusions = [
            {
                id: 1,
                date: new Date('2020-10-03T21:58:26'),
                userWeigth: 82,
                factor: {
                    brand: 'Baxter',
                    unity: 1000,
                    lot: 'ABC123'
                },
                isHemarthrosis: true,
                isBleeding: true,
                isTreatmentContinuation: true,
                bleedingLocal: 'Cotovelo D',
                treatmentLocal: 'Casa'
            },
            {
                id: 2,
                date: new Date('2020-10-03T20:58:26'),
                userWeigth: 82,
                isHemarthrosis: true,
                isBleeding: true,
                isTreatmentContinuation: true,
                bleedingLocal: 'Joelho E',
                treatmentLocal: 'Casa',
            }
        ];

        const tree = create(<InfusionsList infusions={infusions} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
