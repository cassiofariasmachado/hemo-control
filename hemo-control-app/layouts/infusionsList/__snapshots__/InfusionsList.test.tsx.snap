// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfusionsList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 50,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "bleedingLocal": "Cotovelo D",
          "date": 2020-10-04T00:58:26.000Z,
          "factor": Object {
            "brand": "Baxter",
            "lot": "ABC123",
            "unity": 1000,
          },
          "id": 1,
          "isBleeding": true,
          "isHemarthrosis": true,
          "isTreatmentContinuation": true,
          "treatmentLocal": "Casa",
          "userWeigth": 82,
        },
        Object {
          "bleedingLocal": "Joelho E",
          "date": 2020-10-03T23:58:26.000Z,
          "id": 2,
          "isBleeding": true,
          "isHemarthrosis": true,
          "isTreatmentContinuation": true,
          "treatmentLocal": "Casa",
          "userWeigth": 82,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "stretch",
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderRadius": 5,
                "elevation": 3,
                "height": 80,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginHorizontal": 30,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
                "shadowOpacity": 0.2,
                "shadowRadius": 2.2,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Cotovelo D
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Data e local: 
            </Text>
            Casa, 2020-10-3 21:58:26
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Fator: 
            </Text>
             Baxter, 1000 UI, Lote ABC123
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "stretch",
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderRadius": 5,
                "elevation": 3,
                "height": 80,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginHorizontal": 30,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
                "shadowOpacity": 0.2,
                "shadowRadius": 2.2,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Joelho E
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Data e local: 
            </Text>
            Casa, 2020-10-3 20:58:26
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Fator: 
            </Text>
            NÃ£o informado
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
